### ISSUE TITLE
Benchmark results difference

### ISSUE THREAD

**Issue by DavideStenner** [somepago/saint#1]
Hi,
I was trying to reproduce the benchmark (xgboost and lightgbm) results but i can't get the same showed in your paper.

I used this to split the dataset in train, valid and test:
https://github.com/somepago/saint/blob/e0ee763b0c23136ae03fa7cff71ff3b2ce4ba647/data.py#L190

I used early stop on validation and collect test performance as final results and rerun the experiment on 5 different seed (0, 1, ..., 5) as you do for Saint model.

I used standard parameter for xgboost and lightgbm with some regularization.

I used the dataset you provide in the following link: 
https://drive.google.com/file/d/1mJtWP9mRP0a10d1rT6b3ksYkp4XOpM0r/view?usp=sharing

The results i get are:

| Model\Dataset| Bank | Blastchar | arrhytmia | Arcene | Forest | Shoppers | Income | Volkert |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| lightgbm  |  93.46  | 83.71 | 93.18 | 85.25| 99.79 | 93.23 | 92.03 | 71.46 |
| xgboost  |  93.41  | 83.67 | 93.13 | 87.66 | 99.71 | 92.62 | 92.36 | 70.32 |

My experiment show clear improvement of the benchmark result as showed below:

| Model\Dataset| Bank | Blastchar | arrhytmia | Arcene | Forest | Shoppers | Income | Volkert |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
| lightgbm  |  +0.069  | +0.54 | +4.45 | +4.2| +6.5 | +0.03 | -0.54 | +3.55 |
| xgboost  |  +0.45  | +1.89 | +11.14 | +6.25 | +4.38 | +0.11 | 0.05 | +1.37 |

Can you share the code used to calculate benchmark results?

I used also quite standard parameter to train xgboost and lightgbm:

xgboost: 
    - max_depth=8,
    - learning_rate=0.01,
    - tree_method = 'hist',
    - subsample=0.75,
    - colsample_bytree=0.75,
    - reg_alpha= 0.5,
    - reg_lambda= 0.5,

lightgbm:
	- learning_rate= 0.01,
	- max_depth= -1,
	- num_leaves= 2**8,
	- lambda_l1= 0.5,
	- lambda_l2= 0.5,
	- feature_fraction= 0.75,
	- bagging_fraction= 0.75,
	- bagging_freq = 1,

I think i can improve these results by tuning these parameter more.

**Comment by somepago** [comment]
Hello, we have not done any exhaustive hyperparameter search on either the baselines nor our model SAINT since we did not want to overfit the model to the data. We did not use any early stopping nor any sort of regularization for the boosting methods. 

The scores you have posted above, are they the mean AUC scores or the best of the 5 runs?

**Comment by somepago** [comment]
Also thanks for sharing the numbers. We will do hyperparameter tuning on both baselines as well as our model in the next arxiv update of the paper.

**Comment by DavideStenner** [comment]
> Thanks for doing this hyperparameter tuning!  We did not conduct much tuning either for gradient boosting methods or for SAINT (in terms of hyperparameters or regularizers), and we use the numbers from original papers when they exist.  In fact, we did not try different regularizers on SAINT at all.  We definitely plan to spend some time hyperparameter tuning at both for our own method and for competing ones, and we will include these in our next arXiv update. The hyperparameters we used for XGBoost in the paper are same as the ones in VIME paper - [link](https://github.com/jsyoon0823/VIME/blob/996c58cf4c570061b30c38ecf2a754a9af85aafd/supervised_models.py#L72)
> 
> The scores you have posted above, **can you confirm if they are the mean AUC scores or the best of the 5 runs?** In our paper, we have shared the average numbers.
> 
> For Arrythmia and Arcene datasets, we would like to point out that the standard error is high, which we discussed in the supplementary material. Also we would like to point out that, many of the new numbers on the boosting methods are still lower than SAINT’s performance.

Yes the performance are the mean AUC score. 
I think that it is necessary to use the early stop to avoid under fit and over fit. Early stop is not part of regularization but it's standard way to train lightgbm/xgboost. Standard deviation is really high for this reason, because you're training for 100 round and not maybe 800 which are the round needed for the model to converge.
It's quite standard to use a learning rate of 0.05, infinite number of round and early stop.
In my case the standard deviation was lower than your case.

**Comment by somepago** [comment]
We will do an extensive hyper parameter search with early stopping in the next arxiv update. Closing the issue for now since it is not related to the code.

**Comment by 7568** [comment]
> Hi, I was trying to reproduce the benchmark (xgboost and lightgbm) results but i can't get the same showed in your paper.
> 
> I used this to split the dataset in train, valid and test:
> 
> https://github.com/somepago/saint/blob/e0ee763b0c23136ae03fa7cff71ff3b2ce4ba647/data.py#L190
> 
> I used early stop on validation and collect test performance as final results and rerun the experiment on 5 different seed (0, 1, ..., 5) as you do for Saint model.
> 
> I used standard parameter for xgboost and lightgbm with some regularization.
> 
> I used the dataset you provide in the following link: https://drive.google.com/file/d/1mJtWP9mRP0a10d1rT6b3ksYkp4XOpM0r/view?usp=sharing
> 
> The results i get are:
> 
> Model\Dataset	Bank	Blastchar	arrhytmia	Arcene	Forest	Shoppers	Income	Volkert
> lightgbm	93.46	83.71	93.18	85.25	99.79	93.23	92.03	71.46
> xgboost	93.41	83.67	93.13	87.66	99.71	92.62	92.36	70.32
> My experiment show clear improvement of the benchmark result as showed below:
> 
> Model\Dataset	Bank	Blastchar	arrhytmia	Arcene	Forest	Shoppers	Income	Volkert
> lightgbm	+0.069	+0.54	+4.45	+4.2	+6.5	+0.03	-0.54	+3.55
> xgboost	+0.45	+1.89	+11.14	+6.25	+4.38	+0.11	0.05	+1.37
> Can you share the code used to calculate benchmark results?
> 
> I used also quite standard parameter to train xgboost and lightgbm:
> 
> xgboost: - max_depth=8, - learning_rate=0.01, - tree_method = 'hist', - subsample=0.75, - colsample_bytree=0.75, - reg_alpha= 0.5, - reg_lambda= 0.5,
> 
> lightgbm: - learning_rate= 0.01, - max_depth= -1, - num_leaves= 2**8, - lambda_l1= 0.5, - lambda_l2= 0.5, - feature_fraction= 0.75, - bagging_fraction= 0.75, - bagging_freq = 1,
> 
> I think i can improve these results by tuning these parameter more.

hi , can you show your code , because I use your params can not get your result,thanks
